### PR TITLE
add example for controlled components

### PR DESCRIPTION
**Pull request by m-raschle**
the alternative for custom input fields when useRegister can't be used.

### CODE REVIEW

File: docs/docs/forms.md [hunk 1]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
 }
 ```
 
+### Controlled Component Fields
+
+If you're working with a fully-loaded component library, or bring your own production-ready components you may want them to integrate with Redwood seemlessly.
+
+You can create these custom fields through the use of Redwood's `useErrorStyles` and React Hook Form's `Controller`. 
+
+The following example shows a component from [`primereact`](https://www.primefaces.org/primereact/) which can be used in Forms like any of the named input fields (listed above).
```
Comment (jtoar) [inline docs/docs/forms.md:392]:
```suggestion
If you're working with a fully-featured component library, or have your own production-ready components, you may want to integrate them with Redwood's forms seamlessly.
You can via Redwood forms' `useErrorStyles` hook and React Hook Form's `Controller` component.
The following example shows how you could go about integrating a component from [PrimeReact](https://www.primefaces.org/primereact/) for use in in Redwood's forms like any of the named-input fields listed above:
```

File: docs/docs/forms.md [hunk 2]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
...
+
+```jsx
+import { ToggleButton, ToggleButtonProps } from 'primereact/togglebutton'
```
Comment (jtoar) [inline docs/docs/forms.md:395]: Just a personal preference:
```suggestion
import { ToggleButton } from 'primereact/togglebutton'
import type { ToggleButtonProps } from 'primereact/togglebutton'
```

File: docs/docs/forms.md [hunk 3]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
...
+
+import { Controller, RegisterOptions, useErrorStyles } from '@redwoodjs/forms'
+
+interface Prop extends ToggleButtonProps {
```
Comment (jtoar) [inline docs/docs/forms.md:399]:
```suggestion
interface Props extends ToggleButtonProps {
```

File: docs/docs/forms.md [hunk 4]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
...
+  validation?: RegisterOptions
+  errorClassName?: string
+}
+
+const ToggleButtonField = (prop: Prop) => {
```
Comment (jtoar) [inline docs/docs/forms.md:404]:
```suggestion
const ToggleButtonField = (props: Props) => {
```

File: docs/docs/forms.md [hunk 5]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
...
+  const {
+    name,
+    className,
+    errorClassName,
+    defaultValue,
+    validation,
+    style,
+    ...propRest
+  } = prop
```
Comment (jtoar) [inline docs/docs/forms.md:413]:
```suggestion
    ...propsRest
  } = props
```

File: docs/docs/forms.md [hunk 6]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
...
+    })
+
+  return (
+    <Controller
+      name={name}
+      defaultValue={defaultValue}
+      rules={validation}
+      render={({ field: { onChange, onBlur, value, name, ref } }) => (
+        <ToggleButton
+          {...propRest}
```
Comment (jtoar) [inline docs/docs/forms.md:429]:
```suggestion
          {...propsRest}
```

File: docs/docs/forms.md [hunk 7]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
...
+          className={componentClassName}
+          style={{ ...componentStyle, ...style }}
+        />
+      )}
+    />
+  )
+}
+
+export default ToggleButtonField
+
```
Comment (jtoar) [inline docs/docs/forms.md:444]:
```suggestion
```

File: docs/docs/forms.md [hunk 8]
```diff
@@ -383,6 +383,67 @@ const RequiredField = ({ label, name, validation }) => {
...
 
+### Controlled Component Fields
+
+If you're working with a fully-loaded component library, or bring your own production-ready components you may want them to integrate with Redwood seemlessly.
+
+You can create these custom fields through the use of Redwood's `useErrorStyles` and React Hook Form's `Controller`. 
+
+The following example shows a component from [`primereact`](https://www.primefaces.org/primereact/) which can be used in Forms like any of the named input fields (listed above).
+
+```jsx
```
Comment (jtoar) [inline docs/docs/forms.md:394]: Could you change this to: ```tsx title="web/src/components/ToggleButtonField.tsx"

GitHub's suggestion feature can't handle the backticks

**Review by jtoar — CHANGES_REQUESTED** [review]
@m-raschle thanks this is awesome! Went through it and made a few suggestions. Let me know if anything's unclear. The code worked great, just tweaked things for readability and sometimes just for personal preference. But otherwise looks good

File: docs/docs/forms.md (hunk 7 above)
Comment (jtoar) [reply docs/docs/forms.md:444]: (just removing the empty space at the bottom of the code block)

File: docs/docs/forms.md (hunk 5 above)
Comment (jtoar) [reply docs/docs/forms.md:413]: Personal preference here, I've just always seen it pluralized

File: docs/docs/forms.md (hunk 2 above)
Comment (jtoar) [reply docs/docs/forms.md:395]: Another personal preference, just helps upfront for readability to say it's a type

File: docs/docs/forms.md (hunk 5 above)
Comment (m-raschle) [reply docs/docs/forms.md:412]: I usually argue that the pluralized form should be used if the arguments contain multiple objects to clarify that the props can be destructed multiple times. This is definitely the case here. Thanks for your feedback.

File: docs/docs/forms.md [hunk 9]
```diff
@@ -383,6 +383,65 @@ const RequiredField = ({ label, name, validation }) => {
 }
 ```
 
+### Controlled Component Fields
+
+If you're working with a fully-featured component library, or have your own production-ready components, you may want to integrate them with Redwood's forms seamlessly.
+You can via Redwood forms' `useErrorStyles` hook and React Hook Form's `Controller` component.
+The following example shows how you could go about integrating a component from [`primereact`](https://www.primefaces.org/primereact/) for use in in Redwood's forms like any of the named-input fields listed above:
+
+```jsx title="web/src/components/ToggleButtonField/ToggleButtonField.tsx"
```
Comment (jtoar) [inline docs/docs/forms.md:392]:
```suggestion
```tsx title="web/src/components/ToggleButtonField/ToggleButtonField.tsx"
```

**Review by jtoar — APPROVED** [review]
Thanks @m-raschle! This is a great addition, let's get it merged.

**Pull request merged**